docs/proposals/custom-sidecar-config.md

```diff
@@ -0,0 +1,32 @@
+# Custom Container Configuration
+
+## Introduction
+Users and field team have requested access to custom configuration of kubernetes sidecar and init containers. See [#3925](https://github.com/kumahq/kuma/issues/3925). This intuitively aligns with the ProxyTemplate configuration, which provides a free-form configuration of Envoy. The current plan is to introduce a new CRD, similar to ProxyTemplate, which can accommodate sidecar config.
```

Comment:
I'd advice for using [Semantic linefeed](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
2 reasons:
- reviews are easier (can review lines individually)
- (not for you) helps non native English speakers keep sentences short

Reply:
This is a great idea.

Reply:
Although, it does make the text in the md file itself a little less readable. I think on balance I like it though. A comment on a big blob of text is never great.